### PR TITLE
Add JSON Schema generator from IR field metadata

### LINKED ISSUE #733: 1.7.1: Core JSON Schema generator from IR field metadata
## Summary

Create `src/protean/ir/generators/schema.py` — pure functions that convert IR element dicts into standard JSON Schema (Draft 2020-12) dicts with `x-protean-*` extension metadata.

**Parent epic**: #732 (1.7 Schema Generation)

### Core generator function
- `generate_element_schema(element_ir, *, element_type, fqn, aggregate_fqn, ...)` → JSON Schema dict
- `generate_all_schemas(ir)` → `dict[fqn, schema_dict]` walks full IR

### IR field kind → JSON Schema type mapping

| IR Kind | IR Type | JSON Schema |
|---------|---------|-------------|
| `standard` | `String` | `{"type": "string", "maxLength": ..., "minLength": ...}` |
| `standard` | `Integer` | `{"type": "integer", "minimum": ..., "maximum": ...}` |
| `standard` | `Float` | `{"type": "number"}` |
| `standard` | `Boolean` | `{"type": "boolean"}` |
| `standard` | `Date` | `{"type": "string", "format": "date"}` |
| `standard` | `DateTime` | `{"type": "string", "format": "date-time"}` |
| `auto` | `Auto` | `{"type": "string", "format": "uuid"}` |
| `status` | — | `{"type": "string", "enum": [...]}` |
| `value_object` | — | `{"$ref": "#/$defs/VOName"}` |
| `has_one` | — | `{"$ref": "#/$defs/EntityName"}` |
| `has_many` | — | `{"type": "array", "items": {"$ref": "#/$defs/EntityName"}}` |
| `value_object_list` | — | `{"type": "array", "items": {"$ref": "#/$defs/VOName"}}` |
| `reference` | — | `{"type": "string"}` (FK value) |
| `list` | `List` | `{"type": "array", "items": ...}` |
| `dict` | `Dict` | `{"type": "object"}` |

### Protean metadata via `x-protean-*` extensions
```json
{
  "x-protean-element-type": "event",
  "x-protean-aggregate": "ecommerce.ordering.Order",
  "x-protean-fqn": "ecommerce.ordering.OrderPlaced",
  "x-protean-version": 2,
  "x-protean-published": true
}
```

### Additional requirements
- Nested VO/entity resolution via `$defs` with `$ref`
- Optional fields use `anyOf` with `null` (Pydantic convention)
- Deterministic output: sorted keys everywhere
- Follows Pydantic's JSON Schema conventions

## Tests

- Unit tests for each field kind mapping
- Nested VOs and entities (`$defs` + `$ref`)
- Metadata envelope correctness
- Determinism (same IR → identical schema dict)
- `anyOf` for optional fields

## Key files
- Create: `src/protean/ir/generators/schema.py`, `tests/ir/test_schema_generator.py`
- Reference: `src/protean/ir/builder.py:84-200`, `src/protean/ir/generators/base.py`

### PR DESCRIPTION
**Pull request by subhashb**
## Summary

Closes #733

- Add `protean.ir.generators.schema` module with pure functions (`generate_element_schema`, `generate_schemas`) that convert IR element dicts into standard JSON Schema (Draft 2020-12) dicts
- Support all data-carrying domain elements: aggregates, entities, value objects, commands, events, and projections
- Map all IR field kinds to JSON Schema types with constraints (maxLength, minLength, minimum, maximum, enum)
- Resolve nested value objects and entities via `$defs`/`$ref`
- Wrap optional fields with `anyOf` null (Pydantic convention)
- Emit `x-protean-*` extension metadata (element-type, fqn, aggregate, version, published, fact-event, event-sourced, identity-field)
- Ensure deterministic output with sorted keys throughout

## Test plan

- [x] 72 unit tests covering all field type mappings (String, Integer, Float, Boolean, Date, DateTime, Auto, Status, List, Dict, ValueObject, HasOne, HasMany, Reference, ValueObjectList)
- [x] Nested `$defs`/`$ref` resolution for value objects and entities
- [x] `anyOf` wrapping for optional fields
- [x] `required` array correctness
- [x] JSON Schema constraint mapping (maxLength, minLength, minimum, maximum, enum)
- [x] `x-protean-*` extension metadata for all element types
- [x] Default value handling (literal defaults included, callable defaults omitted)
- [x] Field-level descriptions
- [x] `generate_schemas()` full IR processing including projections
- [x] Determinism (same IR → identical schema dict)
- [x] Edge cases (empty elements, missing fields, refs without lookup index)
- [x] Integration test against rich domain with all element types
- [x] 97% code coverage on the new module

### CODE REVIEW

File: tests/ir/test_schema_generator.py [hunk 1]
```diff
@@ -0,0 +1,792 @@
...
+                    flat[fqn] = elem
+
+        element = _find_element(ir, "Order")
+        self.schema = generate_element_schema(element, all_elements=flat)
+
+    def test_value_object_ref(self):
+        p = self.schema["properties"]["shipping_address"]
+        assert "$ref" in p
+        assert p["$ref"] == "#/$defs/ShippingAddress"
+
```
[the remark this answers is not in the document]
Comment (subhashb) [reply tests/ir/test_schema_generator.py]: Good catch. Fixed — optional `$ref` fields (value_object, has_one) are now wrapped in `anyOf: [{$ref: ...}, {type: null}]` consistently with all other optional fields. Tests updated accordingly.

File: src/protean/ir/generators/schema.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+                "properties": dict(sorted(nested_props.items())),
+                "type": "object",
+            }
+            if nested_required:
+                nested_schema["required"] = nested_required
+            defs[ref_name] = dict(sorted(nested_schema.items()))
+
+            # Recurse into nested refs
+            nested_defs = _collect_defs(target_fields, all_elements)
+            defs.update(nested_defs)
```
[the remark this answers is not in the document]
Comment (subhashb) [reply src/protean/ir/generators/schema.py]: Good catch. Added a shared `_visited` set (keyed by target FQN) that is passed through recursive calls. This prevents infinite recursion on cyclic references (e.g., A → B → A) and avoids expanding the same definition more than once. Added a dedicated test exercising this scenario.
